space_packet_parser/xtce/validation.py

```diff
@@ -0,0 +1,441 @@
+            param_ref = entry.get("parameterRef")
+            if param_ref:
+                referenced_parameters.add(param_ref)
+                if param_ref not in parameters:
+                    result.add_error(
+                        f"SequenceContainer references nonexistent Parameter '{param_ref}'",
+                        "MISSING_PARAMETER_REFERENCE",
+                        xpath_location=f"//xtce:ParameterRefEntry[@parameterRef='{param_ref}']",
+                    )
+
+        # Check BaseContainer references to SequenceContainers
+        base_containers = root.xpath("//xtce:BaseContainer", namespaces=namespaces)
+        for base_container in base_containers:
+            container_ref = base_container.get("containerRef")
+            if container_ref and container_ref not in containers:
+                result.add_error(
+                    f"BaseContainer references nonexistent SequenceContainer '{container_ref}'",
+                    "MISSING_CONTAINER_REFERENCE",
+                    xpath_location=f"//xtce:BaseContainer[@containerRef='{container_ref}']",
+                )
+
+        # Check for unused ParameterTypes
+        unused_parameter_types = parameter_types - referenced_parameter_types
+        for unused_type in unused_parameter_types:
+            result.add_error(
+                f"ParameterType '{unused_type}' is defined but never used",
+                "UNUSED_PARAMETER_TYPE",
+                xpath_location=f"//xtce:*[@name='{unused_type}']",
+            )
+
+        # Check for unused Parameters
+        unused_parameters = parameters - referenced_parameters
+        for unused_param in unused_parameters:
+            result.add_error(
+                f"Parameter '{unused_param}' is defined but never used",
+                "UNUSED_PARAMETER",
+                xpath_location=f"//xtce:Parameter[@name='{unused_param}']",
+            )
+
+    except Exception as e:
+        result.add_error(f"Error during structural validation: {e}", "STRUCTURAL_VALIDATION_ERROR")
+    finally:
+        result.validation_time_ms = (time.time() - start_time) * 1000
+
+    return result
+
+
+def validate_xtce(
+    xml_source: Union[str, Path, ElementTree.ElementTree],
+    level: str = "schema",
+    timeout: int = 30,
+) -> ValidationResult:
+    """Validate an XTCE XML document.
+
+    This is the main validation entry point for XTCE documents. It can perform
+    schema or structural validation based on the level parameter.
+
+    Parameters
+    ----------
+    xml_source : Union[str, Path, ElementTree.ElementTree]
+        Path to XML file, XML string content, or ElementTree
+    level : str
+        Validation level: "schema", "structure", or "all"
+    timeout : int
+        Timeout in seconds for schema downloads
+
+    Returns
+    -------
+    ValidationResult
+        Validation results with errors, warnings, and metadata
+    """
+    try:
+        validation_level = ValidationLevel(level.lower())
```

Comment:
This seems like you should just raise immediately and not do this add_error business. If someone isn't using the function with the right calling arguments, they won't want to go and need to look into the error attributes to see something errored.